include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -172,24 +175,40 @@ MGTransferPrebuilt<VECTOR>::copy_to_mg (
 {
   reinit_vector(mg_dof_handler, component_to_block_map, dst);
   bool first = true;
+#ifdef DEBUG_OUTPUT
+  std::cout << "copy_to_mg src " << src.l2_norm() << std::endl;
+  MPI_Barrier(MPI_COMM_WORLD);
+#endif
   for (unsigned int level=mg_dof_handler.get_tria().n_global_levels(); level != 0;)
     {
       --level;
       VECTOR &dst_level = dst[level];
 
+#ifdef DEBUG_OUTPUT
+      MPI_Barrier(MPI_COMM_WORLD);
+#endif
+
       typedef std::vector<std::pair<types::global_dof_index, types::global_dof_index> >::const_iterator IT;
       for (IT i= copy_indices[level].begin();
            i != copy_indices[level].end(); ++i)
         dst_level(i->second) = src(i->first);
 
-      // for (IT i= copy_indices_to_me[level].begin();
-      //      i != copy_indices_to_me[level].end(); ++i)
-      //   dst_level(i->second) = src(i->first);
+      for (IT i= copy_indices_global_mine[level].begin();
+           i != copy_indices_global_mine[level].end(); ++i)
+        dst_level(i->second) = src(i->first);
```

Comment:
And the reason of having both `copy_indices_global_mine` and `copy_indices_level_mine` is that you can only write remote entries but not read remote entries, right?